sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEPropagateEmptyRelation.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.adaptive
+
+import org.apache.spark.sql.catalyst.optimizer.PropagateEmptyRelationBase
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+import org.apache.spark.sql.execution.joins.HashedRelationWithAllNullKeys
+
+/**
+ * Rule [[PropagateEmptyRelationBase]] at AQE side.
+ */
+object AQEPropagateEmptyRelation extends PropagateEmptyRelationBase {
+
+  override protected def checkRowCount(plan: LogicalPlan, hasRow: Boolean): Option[Boolean] = {
+    Some(plan match {
```

Comment:
Can we revert to the previous version? Returning None means we don't know if it's empty or not and should not do optimization based on it.

Reply:
https://github.com/apache/spark/pull/32602/files#diff-b4320e1fb79936232c73e55f34b8a14486b0351a435eeb4914d81a8a756b6afbL58